aten/src/ATen/core/TensorBase.h

```diff
@@ -741,7 +741,7 @@ class TORCH_API TensorBase {
     return *this;
   }
   bool requires_grad() const {
-    return impl_->requires_grad();
+    return impl_->requires_grad() && (is_floating_point() || is_complex());
```

Comment:
In the case of CSR tensor `crow_indices()` return value (that is a tensor with integer dtype), `impl_->requires_grad()` may return `true` when `impl_->autograd_meta_->requires_grad()` returns true.

So, an alternative solution is to make sure that `autograd_meta_` is not set for `crow_indices()` return value. Please advise if this would be a preferred solution and how to achieve this.

^ @albanD

Reply:
Nevermind, adding
```
- name: crow_indices(Tensor(a) self) -> Tensor(a)
  output_differentiability: [False]
```
to derivatives.yaml resolved the problem.